integration-tests/src/tests/nearcore/sync_state_nodes.rs

```diff
@@ -530,3 +535,141 @@ fn sync_state_dump() {
+                            store.clone(),
+                            &genesis,
+                            epoch_manager.clone(),
+                        ) as Arc<dyn RuntimeAdapter>;
+                        (tmp_dir, store, epoch_manager, runtime)
+                    })
+                    .collect::<Vec<(
+                        tempfile::TempDir,
+                        Store,
+                        Arc<EpochManagerHandle>,
+                        Arc<dyn RuntimeAdapter>,
+                    )>>();
+
+            let stores = env_objects.iter().map(|x| x.1.clone()).collect();
+            let epoch_managers = env_objects.iter().map(|x| x.2.clone()).collect();
+            let runtimes = env_objects.iter().map(|x| x.3.clone()).collect();
+
+            let mut env = TestEnv::builder(chain_genesis)
+                .clients_count(num_clients)
+                .stores(stores)
+                .epoch_managers(epoch_managers)
+                .runtimes(runtimes)
+                .use_state_snapshots()
+                .build();
+
+            let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+            let mut blocks = vec![genesis_block.clone()];
+            let signer =
+                InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+            let target_height = epoch_length + 1;
+            for i in 1..=target_height {
+                let block = env.clients[0].produce_block(i).unwrap().unwrap();
+                blocks.push(block.clone());
+                if (i % epoch_length) != 0
+                    && epoch_length - (i % epoch_length) <= num_last_chunks_missing
+                {
+                    // Don't produce chunks for the last blocks of an epoch.
+                    env.clients[0]
+                        .process_block_test_no_produce_chunk(
+                            MaybeValidated::from(block.clone()),
+                            Provenance::PRODUCED,
+                        )
+                        .unwrap();
+                    tracing::info!(
+                        "Block {i}: {:?} -- produced no chunk",
+                        block.header().epoch_id()
+                    );
+                } else {
+                    env.process_block(0, block.clone(), Provenance::PRODUCED);
+                    tracing::info!(
+                        "Block {i}: {:?} -- also produced a chunk",
+                        block.header().epoch_id()
+                    );
+                }
+                env.process_block(1, block, Provenance::NONE);
+
+                let tx = SignedTransaction::send_money(
+                    i + 1,
+                    "test0".parse().unwrap(),
+                    "test1".parse().unwrap(),
+                    &signer,
+                    1,
+                    *genesis_block.hash(),
+                );
+                assert_eq!(env.clients[0].process_tx(tx, false, false), ProcessTxResponse::ValidTx);
```

Comment:
I think doing `env.clients[0].process_tx()` here means this transaction will actually be included in the next block not this one, is this intended?